generate-breaking-changes-doc/action.yaml

```diff
@@ -0,0 +1,103 @@
+name: generate breaking changes document
+description: GitHub Action that creates a breaking changes document based on the configured template.
+inputs:
+  checkout-repo:
+    required: false
+    description: Perform checkout as first step of action
+    default: "true"
+  github-token:
+    required: true
+    description: GitHub token that can checkout the consumer repository as well push against it. e.g. 'secrets.GITHUB_TOKEN'
+  template-url:
+    required: false
+    description: Breaking changes document template URL
+    default: https://raw.githubusercontent.com/open-turo/standards-terraform/main/templates/breaking-changes-doc-template.md
+  validate-against-template:
+    required: false
+    description: Validate that the document was changed from what is in the template
+    default: "true"
+outputs:
+  document-path:
+    description: The relative path to the generated document
+    value: ${{ steps.check.outputs.contains-breaking }}
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      if: inputs.checkout-repo == 'true'
+      with:
+        fetch-depth: 0
+
+    - name: Get current and next major versions
+      id: versions
+      shell: bash
+      run: |
+        # Get the latest tag, strip the leading `v` if present, and extract the major version
+        current=$(git describe --tags --abbrev=0 | sed 's/^v//' |  grep -oE '^([0-9]+)')
+        next="$((current + 1))"
+
+        echo "current_major=$current" >> "$GITHUB_OUTPUT"
+        echo "next_major=$next" >> "$GITHUB_OUTPUT"
+
+    - name: Generate document
+      id: generate
+      shell: bash
+      run: |
+        CURRENT_MAJOR=${{ steps.versions.outputs.current_major }}
+        NEXT_MAJOR=${{ steps.versions.outputs.next_major }}
+        OUTPUT_FILE=docs/breaking-changes/v"$NEXT_MAJOR".md
+        curl -s ${{ inputs.template-url }} --output "$OUTPUT_FILE"
+
+        sed -i.bak -r "s/[-_*][-_*]CURRENT[-_*][-_*]/${CURRENT_MAJOR}/g" "$OUTPUT_FILE"
+        sed -i.bak -r "s/[-_*][-_*]NEXT[-_*][-_*]/${NEXT_MAJOR}/g" "$OUTPUT_FILE"
+
+        rm -f "$OUTPUT_FILE".bak
+
+        echo "document-path=$OUTPUT_FILE" >> "$GITHUB_OUTPUT"
+
+    - name: Check for changes
+      uses: dorny/paths-filter@v2
+      id: document
+      with:
+        filters: |
+          doc: &doc
+            - docs/breaking-changes/v${{ steps.versions.outputs.next_major }}.md
+          added:
+            - added: *doc
+          modified:
+            - modified: *doc
+
+    - name: Fail since document needs to be updated
```

Comment:
@gopkri24 the more I think about this, the more I tend to want to remove this part.  It's pretty flaky and doesn't add much value.
For example, if the template is updated later down the line, this fails.
Also, it can be easily "bypassed" by simply adding a space, char, etc.
I think the PR message telling the owner to update it should be enough. Maybe we could check that the owner adds a reaction to the comment or something similar.

Reply:
hmm..I do see it would add little maintenance, I'm ok removing it